### ISSUE TITLE
Rasa_Core predicting wrong actions

### ISSUE THREAD

**Issue by vinayver198** [RasaHQ/rasa_core#1233]
**Rasa Core version**:

**Python version**: 

**Operating system** (windows, osx, ...):

**Issue**:
Hi,

I am building a FAQ bot. In my case , i am using tensorflow embedding pipeline. I am able to get the right intent but  the actions are not right. With rasa_nlu I am also getting 0.9 confidence level. I don't know what is getting wrong.

I have attached the stories sample and also the domain sample

Domain.yml
intents:
    - Q1
    - Q2
    - Q3
    - Q4
    - Q5
    - Q6
    - Q7
    - Q8
    - Q9
    - Q10
    - Q11
    - Q12
    - Q13
    - Q14
    - Q15
templates:
    utter_default:
        - Sorry I didn't get you. Can you rephrase your query.

actions:
    - utter_default
    - actions.utter_greet
    - actions.utter_goodbye
    - actions.utter_Q1
    - actions.utter_Q2
    - actions.utter_Q3
    - actions.utter_Q4
    - actions.utter_Q5
    - actions.utter_Q6
    - actions.utter_Q7
    - actions.utter_Q8
    - actions.utter_Q9
    - actions.utter_Q10
    - actions.utter_Q11
    - actions.utter_Q12
    - actions.utter_Q13
    - actions.utter_Q14
    - actions.utter_Q15

Stories.md
## story 1
* Q1
    - utter_Q1
    - reset_slots
## story 2
* Q2
    - utter_Q2
    - reset_slots
## story 3
* Q3
    - utter_Q3
    - reset_slots
## story 4
* Q4
    - utter_Q4
    - reset_slots
## story 5
* Q5
    - utter_Q5
    - reset_slots
## story 6
* Q6
    - utter_Q6
    - reset_slots
## story 7
* Q7
    - utter_Q7
    - reset_slots
## story 8
* Q8
    - utter_Q8
    - reset_slots
## story 9
* Q9
    - utter_Q9
    - reset_slots
## story 10
* Q10
    - utter_Q10
    - reset_slots
## story 11
* Q11
    - utter_Q11
    - reset_slots
## story 12
* Q12
    - utter_Q12
    - reset_slots
## story 13
* Q13
    - utter_Q13
    - reset_slots
## story 14
* Q14
    - utter_Q14
    - reset_slots
## story 15
* Q15
    - utter_Q15
    - reset_slots

**Content of domain file** (if used & relevant):
```yaml

```

**Comment by EPedrotti** [comment]
Thanks for raising this issue, @wochinge  will get back to you about it soon.

**Comment by stalinsabu** [comment]
Are you sure these utter_Qx are in your actions.py file. Have you tried not using utter_Qx? maybe its the keyword utter_? I am suggesting this presuming that these are custom actions and not just any utter_ action!

Just rename your **custom actions** to something else not starting with `utter_` for example `action_say_Qx` or `action_utter_Qx`

**Comment by wochinge** [comment]
@vinayver198 
The utters in your stories have the form `utter_Qxx` , but in your domain file they are called `action.utter_Qxx`. These seems weird to me.  Can you help clarifying the issue by helping me out on the following questions?

* What is the exact error:
   - is the wrong action executed or
   - or is no action executed?
* Do you want to use custom actions or what is the purpose of your `action.` prefix?
* can you attach a logfile?

**Comment by vinayver198** [comment]
@wochinge 

Hi,
1. Actually the wrong action is executed.
   for ex : if actual intent is Q15 then rasa_nlu predicts intent correctly as Q15 but the rasa_core doesn't predicts the action utter_Q16
2. Also I have changes utter_Qx to action_say_Qx.
3. Yes these are custom actions.

Sample example of my custom action:

class action_say_Qx(Action):
	def name(self):
		return "action_say_Qx"
								
	def run(self, dispatcher, tracker, domain):
		no = 'Qx'
		Data = "some message"				
		dispatcher.utter_message(Data)
		return [AllSlotsReset()]

**Comment by vinayver198** [comment]
1. action_say_Q{x} randomly gets executed.
2. No, for action, there is custom call. For example, for Q1 to Q16 I will be having 16 custom actions as action_say_Q{x} where x ranges from 1 to 16.
3. Content of domain.yml
intents:
- Q1
- Q2
- Q3
- Q4
- Q5
- Q6
- Q7
- Q8
- Q9
- Q10
- Q11
- Q12
- Q13
- Q14
- Q15
- Q16
- Greet
- GoodBye

templates:
utter_default:
- Sorry I didn't get you. Can you rephrase your query.

actions:
- utter_default
    - actions.action_say_greet
    - actions.action_say_goodbye
    - actions.action_say_Q1
    - actions.action_say_Q2
    - actions.action_say_Q3
    - actions.action_say_Q4
    - actions.action_say_Q5
    - actions.action_say_Q6
    - actions.action_say_Q7
    - actions.action_say_Q8
    - actions.action_say_Q9
    - actions.action_say_Q10
    - actions.action_say_Q11
    - actions.action_say_Q12
    - actions.action_say_Q13
    - actions.action_say_Q14
    - actions.action_say_Q15
    - actions.action_say_Q16


4. Also since it is a FAQ bot and I have increased my number of intent to 500.  But I have to train it online again and again to fix the responses by rasa_core. Predictions by rasa_nlu is coming fine. The isssue is in responses. Also I am attaching my stories for reference. 

## story 1
* Q1
    - action_say_Q1
    - reset_slots
## story 2
* Q2
    - action_say_Q2
    - reset_slots
## story 3
* Q3
    - action_say_Q3
    - reset_slots
## story 4
* Q4
    - action_say_Q4
    - reset_slots
## story 5
* Q5
    - action_say_Q5
    - reset_slots
## story 6
* Q6
    - action_say_Q6
    - reset_slots
## story 7
* Q7
    - action_say_Q7
    - reset_slots
## story 8
* Q8
    - action_say_Q8
    - reset_slots
## story 9
* Q9
    - action_say_Q9
    - reset_slots
## story 10
* Q10
    - action_say_Q10
    - reset_slots
## story 11
* Q11
    - action_say_Q11
    - reset_slots
## story 12
* Q12
    - action_say_Q12
    - reset_slots
## story 13
* Q13
    - action_say_Q13
    - reset_slots
## story 14
* Q14
    - action_say_Q14
    - reset_slots
## story 15
* Q15
    - action_say_Q15
    - reset_slots
## story 16
* Q16
    - action_say_Q16
    - reset_slots
## Generated Story -1797128447469880946
* Greet
    - action_say_greet
    - reset_slots
## Generated Story -1797128447469880946
* GoodBye
    - action_say_goodbye
    - reset_slots

5. Since the samples are getting more than 1000. I am using tensorflow_embedding in configuration file.

**Comment by wochinge** [comment]
The actions in your domain file are prefixed with `action.` , e.g. `actions.action_say_Q1` but in your stories you use `action_say_Q1`. Is this just some error in the provided examples? Since it should not even train with your configuration.

And can you please attach me some logfile of your training / running the bot?

**Comment by stalinsabu** [comment]
> @wochinge
> 
> Hi,
> 
> 1. Actually the wrong action is executed.
>    for ex : if actual intent is Q15 then rasa_nlu predicts intent correctly as Q15 but the rasa_core doesn't predicts the action utter_Q16
> 2. Also I have changes utter_Qx to action_say_Qx.
> 3. Yes these are custom actions.
> 
> Sample example of my custom action:
> 
> ```
> class action_say_Qx(Action):
>     def name(self):
>     return "action_say_Qx"
> 							
>     def run(self, dispatcher, tracker, domain):
> 	no = 'Qx'
> 	Data = "some message"				
> 	dispatcher.utter_message(Data)
> 	return [AllSlotsReset()]
> ```

Wait! Are you sure this action.py file you have provided is the exact thing?
Because if it is, just add an indentation to `return "action_say_Qx"` thats the issue as your syntax is wrong and rasa is not able to understand it
You action should like this now
```
class action_say_Qx(Action):
    def name(self):
        return "action_say_Qx"
							
    def run(self, dispatcher, tracker, domain):
	no = 'Qx'
	Data = "some message"				
	dispatcher.utter_message(Data)
	return [AllSlotsReset()]
```
I am suggesting this based on your provided code! If this is not issue then give your logfile and rasa version

**Comment by vinayver198** [comment]
Hi, 
Is it necessary to have equal number of examples for every intent
